### PR TITLE
Update documents regarding master branch and nightly builds.

### PR DESCRIPTION
**Pull request by davidhutchens**
This PR updates the documents to reflect the deletion of the master branch by removing all references to it.

It also better describes how to find the latest Nightly build since we switched to the main branch.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -123,8 +123,11 @@ Note that these builds may be unstable since the main branch is a work in progre
 
 To get nightly downloads, please
 [click here](https://github.com/logisim-evolution/logisim-evolution/actions/workflows/nightly.yml)
-and browse to the last successful run of `Nightly` worker, which should be on top. Note that due to Github internals,
-all files are provided as ZIP archives. You must unzip the downloaded file to get the installation package.
+and browse to the last successful run of the `Nightly` worker.
+It should be the first from the top that took a several minutes to run.
```
Comment (maehne) [inline README.md:127]: "[...] took several minutes to run."?

File: README.md [hunk 2]
```diff
@@ -123,8 +123,11 @@ Note that these builds may be unstable since the main branch is a work in progre
...
+Nightly runs each night, but terminates quickly and does not produce new packages if nothing has changed.
+Note that due to Github internals, all files are provided as ZIP archives.
```
Comment (maehne) [inline README.md:129]: We should use "GitHub" consistently.

File: docs/developers.md [hunk 3]
```diff
@@ -149,8 +149,5 @@ Once it is running without bugs on your local fork, request a *Pull request* by:
 * On the right-hand side select your fork, for example: *head repository: BFH-ktt1/logisim-evolution*
 * On the right-hand side select your branch, for example: *base: bugfixes*
 * On the left-hand side select the main branch *base: main* (**Important:** All pull requests **MUST**
-  be on the [branch `main`](https://github.com/logisim-evolution/logisim-evolution/tree/main) as
-  the [branch `master`](https://github.com/logisim-evolution/logisim-evolution/tree/master) only
-  holds the code of the latest stable release, and we do not allow any external contributions to that
-  particular branch.
+  be on the [branch `main`](https://github.com/logisim-evolution/logisim-evolution/tree/main).
```
Comment (maehne) [inline docs/developers.md:152]: "be made on the branch [...]"?

**Review by maehne — APPROVED** [review]
Thanks for the documentation update @davidhutchens! I have some minor remarks you might want to consider. As you are the native English speaker, I leave you the final judgement.

File: README.md (hunk 1 above)
Comment (davidhutchens) [reply README.md]: I reworded this section to, hopefully, make it clearer.

File: README.md (hunk 2 above)
Comment (davidhutchens) [reply README.md]: Done.

File: docs/developers.md (hunk 3 above)
Comment (davidhutchens) [reply docs/developers.md]: Done.

File: docs/developers.md [hunk 4]
```diff
@@ -117,9 +117,9 @@ To see all available tasks run: `./gradlew tasks --all`
 `Logisim-evolution` is often updated.
 The [branch `main`](https://github.com/logisim-evolution/logisim-evolution/tree/main)
 is the place where all the work on next release happens.
-Once the code reaches the point it is ready for the next public release, it will
-be merged into the [`master` branch](https://github.com/logisim-evolution/logisim-evolution/tree/master) and released.
-But if you want to contribute, or even just see what we are currently working on, checkout the `main` branch
+Once the code reaches the point it is ready for the next public release, the public release will be created from `main`
```
Comment (maehne) [inline docs/developers.md:120]: Can we somehow avoid using twice "public release"? E.g.: "the release packages will be created [...]"

File: docs/developers.md [hunk 5]
```diff
@@ -149,8 +149,5 @@ Once it is running without bugs on your local fork, request a *Pull request* by:
 * On the right-hand side select your fork, for example: *head repository: BFH-ktt1/logisim-evolution*
 * On the right-hand side select your branch, for example: *base: bugfixes*
 * On the left-hand side select the main branch *base: main* (**Important:** All pull requests **MUST**
-  be on the [branch `main`](https://github.com/logisim-evolution/logisim-evolution/tree/main) as
-  the [branch `master`](https://github.com/logisim-evolution/logisim-evolution/tree/master) only
-  holds the code of the latest stable release, and we do not allow any external contributions to that
-  particular branch.
+  be made on the branch [`main`](https://github.com/logisim-evolution/logisim-evolution/tree/main).
```
Comment (maehne) [inline docs/developers.md:152]: Maybe: "be made against [that branch](https://github.com/logisim-evolution/logisim-evolution/tree/main)."

**Review by maehne — APPROVED** [review]
Looks good to me! I just have to minor suggestions that you may consider.

File: docs/developers.md (hunk 4 above)
Comment (davidhutchens) [reply docs/developers.md]: I decided to simplify the sentence.

File: docs/developers.md (hunk 5 above)
Comment (davidhutchens) [reply docs/developers.md]: Done.

**Review by maehne — APPROVED** [review]
Looks good to me! Thanks @davidhutchens!

**Review by MarcinOrlowski — APPROVED** [review]
LGTM

**Pull request merged**